### ISSUE TITLE
Mirrored posts as forwarded do not show up on profile page

### ISSUE THREAD

**Issue by poVoq** [friendica/friendica#6700]
### Expected behavior
Mirroring posts from RSS feeds should show up on the profile page from where they get federated and users can comment on them.

### Actual behavior
The show up in the personal feed more or less correctly (with the "via" notice and DFRN (?)) and you can also self-comment on them, but they do not show on the profile page and are also not send/federated to other users. This is an account set to the "news" type if that makes any difference.

### Steps to reproduce the problem
Enable option to repost comments in admin settings, add a RSS feed as a connection, go to advanced connection options and set mirror post, then wait for new posts to come through RSS that will be automatically reposted.

### Friendica version you encountered the problem
2019.01

### Friendica source (git, zip)
git
### PHP version
7.3
### SQL version
5.6

**Comment by poVoq** [comment]
Updated to 2019-03RC, but still the same issue. Manually reposting content works.

**Comment by MrPetovan** [comment]
Confirmed by an additional source. This concerns the "Mirror as forwarded post" setting, not the "Mirror as own" setting.

**Comment by annando** [comment]
We should consider dropping this "mirror as forwarded post" since this does interfere with the different networks we are posting to and it does makes problems when we (hopefully) will switch to "real" reshares at some point.

**Comment by poVoq** [comment]
I guess you are referring to the Mastodon specific issue and ActivityPub that Hubzilla also seems to have issues with? See: https://framagit.org/hubzilla/addons/issues/134
I think Pleroma accepts these though, maybe this is rather a Mastodon issue?

As a work around I have seen RSS feeds automatically adding a source link to the top of the bottom of the feed so that it prominently appears on pages pulling the feed. Blogger.com for example does that (for other reasons of course). Might be a "good enough" solution to automatically add these to source feed posts to avoid federation issues?

**Comment by annando** [comment]
@poVoq I'm referring to the problem that Reshares under Mastodon are conceptual different to reshares under Diaspora. And the problem is that either Diaspora and Mastodon do need the original post to accept the reshare - but a reshared feed item is neither an original post under Diaspora nor Mastodon.

So it is better to simply remove the "Mirror as forwarded post"  and to just keep the "Mirror as own".

**Comment by tobiasd** [comment]
> So it is better to simply remove the "Mirror as forwarded post" and to just keep the "Mirror as own".

Could you ask about this in the support forum please. Maybe this is ***the*** main use case of the mirror functionality in the reality outside of the dev group and should be kept, even if it leads to such problems. I really don't know, but this is something that should be done in a broader consensus I think.

**Comment by poVoq** [comment]
For my case it is the main and only use-case, but I guess if for technical reasons it would post it as "own" but with a big link to the original source inserted that would be a not so bad compromise.

**Comment by MrPetovan** [comment]
> I'm currently reviewing the way we import feed items to Friendica and export those posts to Mastodon for a Fediverse release aggregator account, and there are oddities ans obsolete behaviors that could be stamped out.

See https://github.com/friendica/friendica/issues/6813#issuecomment-470380726

**Comment by annando** [comment]
@tobiasd the only option that we have is to change this behaviour so that we always share it as one of our own posts but we add some text on top of it, where we refer to the original item.

**Comment by tobiasd** [comment]
I don't think so. If the majority of users does want to keep this feature, we can also keep it and let the other projects adapt to the additional information about the actual author of the posting which is embedded in the meta information of the posting. But this is something the broader community should discuss and not some elitist developer circle in an obscure thread on github.

**Comment by annando** [comment]
Problem is that our current reshare mechanism is unsafe. Anyone can fake a post that looks like a valid reshare. The other systems don't allow this (and surely never will) - since this is a security issue. And we also should get rid of this as well.

And yes, we can discuss this with a broader audience. But I think we have to change this resharing behaviour as soon as possible.

**Comment by annando** [comment]
In the meantime the mirroring was completely rewritten and works now fine.
